packages/web/hooks/queries/assets/use-coin-fiat-value.ts

```diff
@@ -9,13 +10,24 @@ export function useCoinFiatValue(
   vsCurrency = DEFAULT_VS_CURRENCY
 ): PricePretty | undefined {
   const { price } = useCoinPrice(coin);
+  return useMemo(
+    () => mulPrice(coin, price, vsCurrency),
+    [coin, price, vsCurrency]
+  );
+}
 
-  return useMemo(() => {
-    if (!coin || !price) {
-      return undefined;
-    }
+// mulPrice multiplies the amount of a coin by its price to get the final value.
+// It returns the final value as a PricePretty object.
+// Returns undefined if either the amount or price is undefined.
+export function mulPrice(
```

Comment:
Similar response to:
https://github.com/osmosis-labs/osmosis-frontend/pull/3070#discussion_r1558843007

I believe `FiatCurrency` would require import

Reply:
Since it's a type package, there's no worry to add it to the `utils` package, however, like my previous comment. I'll approve this PR and tentatively open a new PR with these improvements.

Reply:
https://linear.app/osmosis/issue/FE-153/reduce-server-package-tech-debt